src/main/java/io/jenkins/infra/repository_permissions_updater/hosting/MavenVerifier.java

```diff
@@ -295,6 +310,59 @@ private void checkSoftwareConfigurationManagementField(Model model, HashSet<Veri
             }
         }
     }
+
+    private void checkDependencyManagement(Model model, HashSet<VerificationMessage> hostingIssues) {
+        boolean hasPluginDependency = model.getDependencies().stream().anyMatch(d ->
+                d.getGroupId().startsWith("org.jenkins-ci") ||
+                        d.getGroupId().startsWith("io.jenkins") ||
+                        d.getGroupId().equals("org.jenkinsci.plugins") ||
+                        d.getGroupId().equals("org.jvnet.hudson.plugins")
+        );
+        DependencyManagement depManagement = model.getDependencyManagement();
+        Optional<Dependency> bom = depManagement.getDependencies().stream().filter(d ->
+                d.getGroupId().equals("io.jenkins.tools.bom") &&
+                        d.getArtifactId().startsWith("bom-")
+        ).findFirst();
+        if (hasPluginDependency && bom.isEmpty()) {
+            hostingIssues.add(new VerificationMessage(VerificationMessage.Severity.WARNING, "You're using one or more Jenkins plugins without making use of the bom. " +
+                    "Please check if those plugin dependencies are covered by the bom and if yes add a corresponding dependencyManagement section to your pom.xml and remove the version for those dependencies." +
+                    " See [here](https://www.jenkins.io/doc/developer/plugin-development/dependency-management/#jenkins-plugin-bom) for details."));
+        }
+        if (hasPluginDependency && bom.isPresent()) {
+            model.getDependencies().stream().filter(d -> d.getGroupId().startsWith("org.jenkins-ci") ||
+                    d.getGroupId().startsWith("io.jenkins") ||
+                    d.getGroupId().equals("org.jenkinsci.plugins") ||
+                    d.getGroupId().equals("org.jvnet.hudson.plugins")).forEach(d -> {
+                if (d.getVersion() != null) {
+                    hostingIssues.add(new VerificationMessage(VerificationMessage.Severity.INFO, "Please check if the dependency `%s:%s` is covered by the bom. If yes then the version should be removed.", d.getGroupId(), d.getArtifactId()));
+                }
+            });
+        }
+        JenkinsVersion jenkinsVersion = getJenkinsVersion(model);
+        if (bom.isPresent() && jenkinsVersion != null) {
+            Dependency b = bom.get();
+            String artifactId = b.getArtifactId();
+            artifactId  = artifactId.replace("${jenkins.baseline}", jenkinsVersion.baseline());
+            String expected = "bom-" + jenkinsVersion.baseline() + ".x";
+            if (!expected.equals(artifactId)) {
+                hostingIssues.add(new VerificationMessage(VerificationMessage.Severity.REQUIRED, "The artifactId `%s` of the bom is not in sync with the jenkins baseline version `%s`", artifactId, jenkinsVersion.baseline()));
+            }
+            if (!model.getProperties().containsKey("jenkins.baseline")) {
```

Comment:
Should we enforce the usage of `jenkins.baseline` even when no dependencyManagement is required?
The archetype for an empty plugin also defines a dependeManagement section and set jenkins.baseline

Reply:
yes I think so

Reply:
We should then also enforce the presence of the dependencyManagement I think

Reply:
why does dependency management need to be there if the plugin has no dependencies?

Reply:
The `jenkins.baseline` is also only really needed when you need the bom. And the archetype for an empty plugin also includes the bom and has no dependencies.
But I'm also fine with only enforcing the `jenkins.baseline` property

Reply:
understood I think fine to leave out though, someone can start from archetype then remove bom if unneeded I don't think it should be a blocker

Reply:
Ok so enforcing the jenkins.baseline now but not the dependencymanagement